### PR TITLE
Print errors to stderr by default, and remove configuration option

### PR DESCRIPTION
**Pull request by greysteil**
### What was the end-user problem that led to this PR?

The problem was #6729 - Bundler unexpectedly outputs error and warning messages to STDOUT.

### What was your diagnosis of the problem?

My diagnosis was that whilst very minorly breaking, this is essentially a bug fix, and should be considered for inclusion for Bundler 2.0 even if very few other breaking changes are.

### What is your fix for the problem, implemented in this PR?

My fix was so switch output for warning and error messages to STDERR, and remove the configuration option (as is redundant once the setup is flipped - anyone wanting to redirect those message to STDOUT could do so in their shell).

### Why did you choose this fix out of the possible options?

I chose this fix because I think the new behaviour is what everyone would expect, and we should get it out from behind its feature switch sooner rather than later.

Alternatively, we might want to keep the Bundler 2.0 release "purer" by only dropping Ruby versions in it - that's totally fine too, but I figured we should have the code to discus #6729, rather than doing it in abstract.

### CODE REVIEW

**Comment by colby-swandale** [comment]
Apologies but i'm confused, the existing code is already set to be enabled for Bundle 2. Why do we need to remove the feature flag?

File: spec/commands/exec_spec.rb [hunk 1]
```diff
@@ -594,17 +579,6 @@ def bin_path(a,b,c)
...
-      let(:exit_code) { 1 }
-      let(:expected) { super() << "\nbundler: failed to load command: #{path} (#{path})" }
-      let(:expected_err) do
-        "RuntimeError: ERROR\n  #{path}:10" +
-          (Bundler.current_ruby.ruby_18? ? "" : ":in `<top (required)>'")
-      end
-      it_behaves_like "it runs"
-    end
-
     context "the executable raises", :bundler => "2" do
```
Comment (deivid-rodriguez) [inline spec/commands/exec_spec.rb]: You should remove `:bundler => 2` so that this test actually runs, always.

File: spec/commands/exec_spec.rb [hunk 2]
```diff
@@ -615,15 +589,6 @@ def bin_path(a,b,c)
...
-    context "the executable raises an error without a backtrace", :bundler => "< 2" do
-      let(:executable) { super() << "\nclass Err < Exception\ndef backtrace; end;\nend\nraise Err" }
-      let(:exit_code) { 1 }
-      let(:expected) { super() << "\nbundler: failed to load command: #{path} (#{path})" }
-      let(:expected_err) { "Err: Err" }
-
-      it_behaves_like "it runs"
-    end
-
     context "the executable raises an error without a backtrace", :bundler => "2" do
```
Comment (deivid-rodriguez) [inline spec/commands/exec_spec.rb]: Same here.

**Comment by deivid-rodriguez** [comment]
@greysteil I wouldn't work on this any more because there's no green light for it, but just in case we end up doing this, documentation about `BUNDLE_ERROR_ON_STDERR` would also need to be removed.

File: spec/commands/exec_spec.rb [hunk 3]
```diff
@@ -570,21 +572,6 @@ def bin_path(a,b,c)
...
-      if LessThanProc.with(RUBY_VERSION).call("1.9")
-        # Kernel#exec in ruby < 1.9 will raise Errno::ENOEXEC if the command content is empty,
-        # even if the command is set as an executable.
-        pending "Kernel#exec is different"
-      else
-        it_behaves_like "it runs"
-      end
-    end
-
     context "the executable is empty", :bundler => "2" do
```
Comment (deivid-rodriguez) [inline spec/commands/exec_spec.rb]: Missed this tag on my initial review that should also be removed...

**Comment by greysteil** [comment]
I have no idea what's causing the final test failure here. I'll have another go at fixing it tomorrow, unless anyone can spot anything I'm missing.

**Comment by greysteil** [comment]
@colby-swandale - any reason I can't base this off of `2-0-stable`?

**Comment by colby-swandale** [comment]
Seeing this is not fixing anything broken on `2-0-0`, this will need to go through the normal release process after 2.0 is released.

**Comment by greysteil** [comment]
Yep, at this point this PR is just code cleanup, and removing the `error_on_stderr` option.

**Comment by deivid-rodriguez** [comment]
I rebased this PR and added some extra simplifications. Thanks @colby-swandale from preventing me from stealing attribution :sweat_smile:, and @greysteil for creating this PR :heart:.

**Comment by greysteil** [comment]
Thanks @deivid-rodriguez! ❤️

**Comment by deivid-rodriguez** [comment]
@greysteil Can't ask for a review from you since you're the author (I'm always bothered by this :grimacing:), but feel free to merge away if it looks good to you! :)

**Comment by greysteil** [comment]
@bundlerbot r+

**Pull request merged**